core/native/src/main/scala/cats/effect/unsafe/KqueueSystem.scala

```diff
@@ -161,13 +161,118 @@ object KqueueSystem extends PollingSystem {
       event.flags = (flags.toInt | EV_ONESHOT).toUShort
 
       callbacks.update(encodeKevent(ident, filter), cb)
+      incrementOperationCount(filter)
 
       changeCount += 1
     }
 
+    private var totalReadSubmitted = 0L
+    private var totalReadSucceeded = 0L
+    private var totalReadErrored = 0L
+    private var totalReadCanceled = 0L
+    private var readOutstanding = 0
+
+    private var totalWriteSubmitted = 0L
+    private var totalWriteSucceeded = 0L
+    private var totalWriteErrored = 0L
+    private var totalWriteCanceled = 0L
+    private var writeOutstanding = 0
+
+    private[this] val pollerMetrics = new PollerMetrics {
+      override def operationsOutstandingCount(): Int = readOutstanding + writeOutstanding
+
+      override def totalOperationsSubmittedCount(): Long =
+        totalReadSubmitted + totalWriteSubmitted
+
+      override def totalOperationsSucceededCount(): Long =
+        totalReadSucceeded + totalWriteSucceeded
+
+      override def totalOperationsErroredCount(): Long = totalReadErrored + totalWriteErrored
+
+      override def totalOperationsCanceledCount(): Long = totalReadCanceled + totalWriteCanceled
+
+      override def acceptOperationsOutstandingCount(): Int = 0
+
+      override def totalAcceptOperationsSubmittedCount(): Long = 0L
+
+      override def totalAcceptOperationsSucceededCount(): Long = 0L
+
+      override def totalAcceptOperationsErroredCount(): Long = 0L
+
+      override def totalAcceptOperationsCanceledCount(): Long = 0L
+
+      override def connectOperationsOutstandingCount(): Int = 0
+
+      override def totalConnectOperationsSubmittedCount(): Long = 0L
+
+      override def totalConnectOperationsSucceededCount(): Long = 0L
+
+      override def totalConnectOperationsErroredCount(): Long = 0L
+
+      override def totalConnectOperationsCanceledCount(): Long = 0L
+
+      override def readOperationsOutstandingCount(): Int = readOutstanding
+
+      override def totalReadOperationsSubmittedCount(): Long = totalReadSubmitted
+
+      override def totalReadOperationsSucceededCount(): Long = totalReadSucceeded
+
+      override def totalReadOperationsErroredCount(): Long = totalReadErrored
+
+      override def totalReadOperationsCanceledCount(): Long = totalReadCanceled
+
+      override def writeOperationsOutstandingCount(): Int = writeOutstanding
+
+      override def totalWriteOperationsSubmittedCount(): Long = totalWriteSubmitted
+
+      override def totalWriteOperationsSucceededCount(): Long = totalWriteSucceeded
+
+      override def totalWriteOperationsErroredCount(): Long = totalWriteErrored
+
+      override def totalWriteOperationsCanceledCount(): Long = totalWriteCanceled
+    }
```

Comment:
Follow-up to https://github.com/typelevel/cats-effect/pull/4365. 
```suggestion
      override def toString: String = "Kqueue"
    }
```